src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections.Generic;
+using FluentValidation.Results;
+using NzbDrone.Core.Tv;
+
+namespace NzbDrone.Core.Notifications.CustomScript
+{
+    public class CustomScript : NotificationBase<CustomScriptSettings>
+    {
+        private readonly ICustomScriptService _customScriptService;
+
+        public CustomScript(ICustomScriptService customScriptService)
+        {
+            _customScriptService = customScriptService;
+        }
+
+        public override string Link
+        {
+            //TODO: Provide a proper link on how to use the custom script
+            get { return "http://wiki.sonarr.tv/"; }
+        }
+
+        public override void OnGrab(string message)
+        {
+        }
+
+        public override void OnDownload(DownloadMessage message)
+        {
+            _customScriptService.OnDownload(message.Series, message.EpisodeFile, Settings);
+        }
+
+        public override void AfterRename(Series series)
+        {
+            //TODO: After rename needs to be optional before we can enable this
+            //If this is enabled after merge we need to enable after rename for everything except for Custom Script
+//            _customScriptService.AfterRename(series, Settings);
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "Custom Script";
+            }
+        }
+
+        public override ValidationResult Test()
+        {
+            var failures = new List<ValidationFailure>();
```

Comment:
No, this is testing whether the notification works (in this case, just checking the script exists), but normally it would send a test message.